File: 0487-max-consecutive-ones-ii/0487-max-consecutive-ones-ii.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int next_zero_flipped = 0, seq_without_zero = 0;
        int res = 0;
        for(int num: nums){
            if(num == 0){
                res = Math.max(res, next_zero_flipped);
                next_zero_flipped = seq_without_zero + 1;
                seq_without_zero = 0;
            }else{
                seq_without_zero +=1;
                next_zero_flipped += 1;
            }
        }
        res = Math.max(next_zero_flipped, res);
        res = Math.max(res, seq_without_zero);
        return res;
    }
}

File: 0487-max-consecutive-ones-ii/0487-max-consecutive-ones-ii.rs
impl Solution {
    pub fn find_max_consecutive_ones(nums: Vec<i32>) -> i32 {
        //Question turns to: can have at most 1 '0' to have a valid string

        let mut left = 0;
        let mut curr = 0;
        let mut res = 0;
        for right in 0..nums.len(){
            if nums[right] == 0{
                curr += 1
            }
            while curr > 1{
                if nums[left] == 0{
                    curr -= 1
                }
                left += 1;
            }
            res = res.max(right-left +1);
        }
        res as i32
    }
}